python/cudf/cudf/core/column/column.py

```diff
@@ -284,13 +284,8 @@ def _concat(cls, objs, dtype=None):
         return col
 
     def dropna(self):
-        dropped_col = libcudf.stream_compaction.drop_nulls([self])
-        if not dropped_col:
-            return column_empty_like(self, newsize=0)
-        else:
-            dropped_col = dropped_col[0]
-            dropped_col.mask = None
-            return dropped_col
+        dropped_col = self.as_frame()._drop_nulls()._data[None].copy(deep=False)
+        return dropped_col
```

Comment:
What API is actually hitting the column level `dropna` now? Ideally we could drop this entire API from `Column` and move it into the `Frame` class instead to be used by `Series` / `Index`

Reply:
We can do this, in that case if anytime if we want to use dropna, it should be converted to series/frame though.

Reply:
Yea we probably don't want to do that. @shwina any ideas on making this cleaner? Maybe we should have a decorator to handle the `as_frame` and `._data[None]` accessor? This looks kinda gross / dangerous from my perspective.

Reply:
Connected offline with @rgsl888prabhu and @shwina and the path forward for cleaning this up:
```
Column.as_frame() --> Frame({None: self.copy(deep=False)})
Frame.as_column() --> self._data[None].copy(deep=False) # also assert only 1 data column, no index, and name is `None`
```

Reply:
🔥